app/src/config/appConfig.ts

```diff
@@ -0,0 +1,19 @@
+interface AppConfig {
+    apiUrl: () => string
+}
+
+const devConfig: AppConfig = {
+    apiUrl: () => "http://localhost:8080"
+};
+
+const prodConfig: AppConfig = {
+    apiUrl: () => `https://${window.location.host}/packit/api`,
```

Comment:
Why not? Unfortunately your suggested approach would only work if the url is known at compile time, but it's only known at the point of deployment.

Reply:
Why would the url be unknown ? isn't that a production url ? 

In production, since the app is dockerized, if we create a docker network named `packit` , we can use the network path as baseUrl i.e `http://packit/api`.

Reply:
1. Packit, like OrderlyWeb, will be deployed to lots of different locations (malaria, Montagu, HIV, etc)
2. The React app makes requests via the browser, so it doesn't have access to internal docker network names, just public urls. These will be things like `malaria.dide.ic.ac.uk` or `montagu.dide.ic.ac.uk`.

Reply:
So I guess window.location.host would work as apiUrl in the devConfig too... but maybe you just want to make placeholder configs here as some config may differ in future.?

Reply:
Yeah true. I was imagining this might grow but unclear if it will.

Reply:
Oh wait, no it wouldn't work locally. It'd come back as `localhost:3000/packit/api` but the api is at `localhost:8080`